tests/integration/test_office365/test_configuration/test_invalid.py

```diff
@@ -0,0 +1,247 @@
+    elif tags_to_apply == 'invalid_interval':
+        wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
+                                callback=callback_detect_interval_err,
+                                accum_results=1,
+                                error_message='Did not receive expected '
+                                              'sched_scan_validate_parameters(): ERROR').result()
+    elif tags_to_apply == 'invalid_curl_max_size':
+        wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
+                                callback=callback_detect_curl_max_size_err,
+                                accum_results=1,
+                                error_message='Did not receive expected '
+                                              'sched_scan_validate_parameters(): ERROR').result()
+    elif tags_to_apply == 'invalid_tenant_id':
+        wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
+                                callback=callback_detect_tenant_id_err,
+                                accum_results=1,
+                                error_message='Did not receive expected '
+                                              'sched_scan_validate_parameters(): ERROR').result()
+    elif tags_to_apply == 'invalid_client_id':
+        wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
+                                callback=callback_detect_client_id_err,
+                                accum_results=1,
+                                error_message='Did not receive expected '
+                                              'sched_scan_validate_parameters(): ERROR').result()
+    elif tags_to_apply == 'invalid_client_secret':
+        wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
+                                callback=callback_detect_client_secret_err,
+                                accum_results=1,
+                                error_message='Did not receive expected '
+                                              'sched_scan_validate_parameters(): ERROR').result()
+    elif tags_to_apply == 'invalid_subscription':
+        wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
+                                callback=callback_detect_subscription_err,
+                                accum_results=1,
+                                error_message='Did not receive expected '
+                                              'sched_scan_validate_parameters(): ERROR').result()
+    else:
+        wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
+                                callback=callback_detect_read_err,
+                                accum_results=1,
+                                error_message='Did not receive expected '
+                                              'wm_gcp_read(): ERROR:').result()
```

Comment:
All this code could be merged if you define a map where the keys are the content of tags_to_apply and the values are the callbacks, then you would just do:

```
wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
                                callback=mymap[tags_to_apply],
                                accum_results=1,
                                error_message='Did not receive expected '
                                              'wm_gcp_read(): ERROR:').result()
```

I think we could do this to make the code easier to read and maintain, but it works perfectly like it is, so this change is up to you!

Reply:
Done!